src/coreclr/gc/env/gcenv.object.h

```diff
@@ -11,7 +11,7 @@ extern bool g_oldMethodTableFlags;
 // ARM requires that 64-bit primitive types are aligned at 64-bit boundaries for interlocked-like operations.
 // Additionally the platform ABI requires these types and composite type containing them to be similarly
 // aligned when passed as arguments.
```

Comment:
```suggestion
// Some platform ABIs require that 64-bit primitive types are aligned at 64-bit boundaries.
```
Places that require this alignment are not limited to interlocked operations and argument passing. There is a whole bunch of other places that depend on this alignment.